examples/abstract-tok/index.rsh

```diff
@@ -0,0 +1,44 @@
+'reach 0.1';
+
+const MT = Maybe(Token);
```

Comment:
I think another good test would be making it so the HumbleSwap code is actually the same but with a `Maybe(Token)` parameter for the A side

Reply:
I don't think it's possible still because `paySpec` requires a hardcoded array of identifiers

Reply:
Could you write `.paySpec( mTokA.match({None: xyz, Some: abc}) )`? I thought that was an evaluated position?

Reply:
No, it's not evaluated. We use the `JSExpression` directly to generate code in `doFork`

Reply:
Well, maybe we should evaluate it and then synthesize something new in its place :) If not, then we should at least write down the examples that fail/are-not-allowed because it is not evaluated